example/lib/main.dart

```diff
@@ -59,7 +59,9 @@ class _RequestHomePageState extends State<RequestHomePage> {
 
   late final store = RequestStore(
     requests: <RequestItem>[],
-    onAddRequest: print,
+    onAddRequest: (item) {
+      return item.copyWith(id: 'new_id');
```

Comment:
Do you think this needs to be async at this point? what happens if there's an error on creation?